### PR TITLE
Hide tooltip when there is no text

### PR DESCRIPTION
**Pull request by dskloetd**
# Motivation

Sometimes we want to have a tooltip conditional on some logic, such as [here](https://github.com/dfinity/nns-dapp/blob/9f0bd72c6e315adf7443bed1ee14571e600cd2a7/frontend/src/lib/components/neuron-detail/NeuronSelectPercentage.svelte#L65).
It would be convenient if the tooltip was just hidden if there is no text.

# Changes

1. Allow `text` to be undefined and hide the tooltip in that case.
2. Reposition the tooltip if the `text` changes.
3. Change the showcase to allow changing the `text`.
4. Add unit tests.

# Screenshots

Looks the same when it's not hidden.

### CODE REVIEW

File: src/routes/(split)/components/tooltip/+page.md [hunk 1]
```diff
@@ -1,19 +1,25 @@
 <script lang="ts">
   import Tooltip from "$lib/components/Tooltip.svelte";
   import Checkbox from "$lib/components/Checkbox.svelte";
+  import Input from "$lib/components/Input.svelte";
 
   let overflowHidden = true;
   let showShowcase = true;
+  let tooltipText = "This button is disabled because of a long and complicated explanation.";
```
Comment (lmuntaner) [inline src/routes/(split)/components/tooltip/+page.md]: Why did you change the text?

File: src/tests/lib/components/Tooltip.spec.ts [hunk 2]
```diff
@@ -18,4 +21,17 @@ describe("Tooltip", () => {
     ).toBeInTheDocument();
     expect(container.querySelector("[id='tid']")).toBeInTheDocument();
   });
+
+  it("should hide without text", () => {
```
Comment (lmuntaner) [inline src/tests/lib/components/Tooltip.spec.ts:25]: Should we also test with `undefined`?

File: src/tests/lib/components/TooltipTest.svelte [hunk 3]
```diff
@@ -1,7 +1,9 @@
 <script>
   import Tooltip from "$lib/components/Tooltip.svelte";
+
+  export let text = "text";
```
Comment (lmuntaner) [inline src/tests/lib/components/TooltipTest.svelte:4]: Shouldn't this be the same type as in the tooltip to allow for `undefined`?

**Review by lmuntaner — COMMENTED** [review]
A couple of comments

File: src/routes/(split)/components/tooltip/+page.md (hunk 1 above)
Comment (dskloetd) [reply src/routes/(split)/components/tooltip/+page.md]: That was an accident that I only realized when updating the screenshot.

File: src/tests/lib/components/Tooltip.spec.ts (hunk 2 above)
Comment (dskloetd) [reply src/tests/lib/components/Tooltip.spec.ts:25]: Done

File: src/tests/lib/components/TooltipTest.svelte (hunk 3 above)
Comment (dskloetd) [reply src/tests/lib/components/TooltipTest.svelte:4]: Done

**Review by lmuntaner — APPROVED** [review]
LGTM!

File: src/lib/components/Tooltip.svelte [hunk 4]
```diff
@@ -29,6 +29,10 @@
       return;
     }
 
+    if (isNullish(text) || text.length === 0) {
```
Comment (mstrasinskis) [inline src/lib/components/Tooltip.svelte:32]: Not really important, but this could be a nice local function like `const notText  = () => isNullish(text) || text.length === 0;` Or even a variable.

Comment (peterpeterparker) [reply src/lib/components/Tooltip.svelte:32]: We have a utils `notEmptyString`.

Comment (dskloetd) [reply src/lib/components/Tooltip.svelte]: Thanks. Done.
